be/src/vec/runtime/vdatetime_value.cpp

```diff
@@ -3812,4 +3812,32 @@ template bool DateV2Value<DateTimeV2ValueType>::datetime_trunc<TimeUnit::YEAR>()
 template bool DateV2Value<DateTimeV2ValueType>::datetime_trunc<TimeUnit::QUARTER>();
 template bool DateV2Value<DateTimeV2ValueType>::datetime_trunc<TimeUnit::WEEK>();
 
+CalcFirstWeekPolicy* CalcFirstWeekPolicy::get_policy_by_code(uint8_t code) {
+    static std::unordered_map<uint8_t, CalcFirstWeekPolicy*> instances;
```

Comment:
when multithreads hits the if branch below, there's data racing problem. this function could be refactor. consider preload it or remove this cache?